connector_equipment_service/models/agreement_serviceprofile.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (C) 2019 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import _, api, models
+
+
+class AgreementServiceProfile(models.Model):
+    _inherit = ['agreement.serviceprofile']
+
+    @api.multi
+    def write(self, vals):
+        # If equipment was empty and now set to managed equipment
+        if vals.get('equipment_id', False) and not self.equipment_id:
+            new = self.env['maintenance.equipment'].\
+                browse(vals.get('equipment_id'))
+            new._connect('add_service', serviceprofiles=self)
+            self.message_post(body=_('Added Service'))
+        # If SP is changed but not the managed equipment
+        if 'equipment_id' not in vals and self.equipment_id.managed:
+            self.equipment_id._connect('update_service',
+                                       serviceprofiles=self)
+            self.message_post(body=_('Updated Service'))
+        # If SP state is changed to In Progress and equipment is managed
+        if vals.get('stage_id', False) == 11 and self.equipment_id.managed:
```

Comment:
You shouldn't use a Record ID.
Use an xml id reference, something like:
`== self.env.ref('module.stage_in_progress').id`